Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyPostedInvoiceExport.Codeunit.al

```diff
@@ -0,0 +1,423 @@
+    end;
+
+    local procedure MapBillInformation(
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary;
+        SalesInvoiceHeader: Record "Sales Invoice Header"
+    )
+    var
+        ShpfyCustomer: Record "Shpfy Customer";
+    begin
+        TempShpfyOrderHeader."Bill-to Name" := CopyStr(SalesInvoiceHeader."Bill-to Name", 1, MaxStrLen(TempShpfyOrderHeader."Bill-to Name"));
+        TempShpfyOrderHeader."Bill-to Name 2" := SalesInvoiceHeader."Bill-to Name 2";
+        TempShpfyOrderHeader."Bill-to Address" := SalesInvoiceHeader."Bill-to Address";
+        TempShpfyOrderHeader."Bill-to Address 2" := SalesInvoiceHeader."Bill-to Address 2";
+        TempShpfyOrderHeader."Bill-to Post Code" := SalesInvoiceHeader."Bill-to Post Code";
+        TempShpfyOrderHeader."Bill-to City" := SalesInvoiceHeader."Bill-to City";
+        TempShpfyOrderHeader."Bill-to County" := SalesInvoiceHeader."Bill-to County";
+        TempShpfyOrderHeader."Bill-to Country/Region Code" := SalesInvoiceHeader."Bill-to Country/Region Code";
+        TempShpfyOrderHeader."Bill-to Customer No." := SalesInvoiceHeader."Bill-to Customer No.";
+
+        ShpfyCustomer.SetRange("Customer No.", SalesInvoiceHeader."Bill-to Customer No.");
+        if ShpfyCustomer.FindFirst() then begin
+            TempShpfyOrderHeader.Email := CopyStr(ShpfyCustomer.Email, 1, MaxStrLen(TempShpfyOrderHeader.Email));
+            TempShpfyOrderHeader."Phone No." := ShpfyCustomer."Phone No.";
+        end;
+    end;
+
+    local procedure MapShipToInformation(
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary;
+        SalesInvoiceHeader: Record "Sales Invoice Header"
+    )
+    begin
+        TempShpfyOrderHeader."Ship-to Name" := CopyStr(SalesInvoiceHeader."Ship-to Name", 1, MaxStrLen(TempShpfyOrderHeader."Ship-to Name"));
+        TempShpfyOrderHeader."Ship-to Name 2" := SalesInvoiceHeader."Ship-to Name 2";
+        TempShpfyOrderHeader."Ship-to Address" := SalesInvoiceHeader."Ship-to Address";
+        TempShpfyOrderHeader."Ship-to Address 2" := SalesInvoiceHeader."Ship-to Address 2";
+        TempShpfyOrderHeader."Ship-to Post Code" := SalesInvoiceHeader."Ship-to Post Code";
+        TempShpfyOrderHeader."Ship-to City" := SalesInvoiceHeader."Ship-to City";
+        TempShpfyOrderHeader."Ship-to County" := SalesInvoiceHeader."Ship-to County";
+        TempShpfyOrderHeader."Ship-to Country/Region Code" := SalesInvoiceHeader."Ship-to Country/Region Code";
+    end;
+
+    local procedure MapCustomerLedgerEntryInformation(
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary)
+    var
+        CustLedgerEntry: Record "Cust. Ledger Entry";
+    begin
+        CustLedgerEntry.Get(SalesInvoiceHeader."Cust. Ledger Entry No.");
+        CustLedgerEntry.CalcFields("Remaining Amount");
+
+        if CustLedgerEntry."Remaining Amount" = 0 then
```

Comment:
SalesInvoiceHeader."Remaining Amount" does the same. 

Map Customer Ledger Entry Info is a bit vague and missleading.
Maybe something like

ShpfyOrderHeader.Unpaid := IsInvoiceUnpaid(SalesInvoiceHeader)

IsInvoiceUnpaid 
    InvoiceHeader.CalcFields(RemainingAmt)
    Exit(RemainingAmt <> 0)